homework

danildemchenko opened the pull request.

danildemchenko committed:
homework

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilCustomView.swift:
```diff
@@ -0,0 +1,76 @@
+    private func addConstraints() {
+        delegateButton.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.width.equalTo(Constants.ButtonSizes.Basics.width)
```
Кстати ты мог бы сделать в константах Constants.Button.basicSize: CGSize = (whidhh: ..., height: ...) и тут использовать $0.size.equalTo(Constants.Button.basicSize). Так должно сработать))) А в др местах если надо Constants.Button.basicSize.width или .height.

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTransferDataController.swift:
```diff
@@ -0,0 +1,62 @@
+import UIKit
+
+protocol DanilTransferDataDelegate: AnyObject {
+    func changeBackgroundColor(backgroundColor: UIColor) -> Void
```
Нейминг плох. Строка должна читаться, как предложение, а что у тебя. "Изменить цвет фона цвет фона 'красный'" - понимаешь? Можно changeBackgroundColor(to color: UIColor) - при вызове будет читаться как: "Изменить цвет фона на красный". Или change(backgroundColorTo color: UIColor) - тоже самое будет) Короче дублирование backgroundColor  точно лишнее

danildemchenko replied:
оке, понял!

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTransferDataController.swift:
```diff
@@ -0,0 +1,62 @@
+        addConstraints()
+        configure()
+        
+        customView.changeBackgroundColorCallback =  {
```
Этот код до закрытия блока стоит перенеcти в configure.

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTransferDataController.swift:
```diff
@@ -0,0 +1,62 @@
+    
+    private func addConstraints() {
+        customView.snp.makeConstraints {
+            $0.center.equalToSuperview()
```
$0.center.width.equalToSuperview() а ниже можно удалить

VAmpersand committed:
Merge branch 'main' into DanilsBranch

danildemchenko committed:
fixes

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilTransitionDemoController.swift:
```diff
@@ -37,8 +37,8 @@ class DanilTransitionDemoController: UIViewController {
         seguePopoverButton.snp.makeConstraints {
-            $0.width.greaterThanOrEqualTo(Constants.ButtonSizes.Basics.width)
-            $0.height.equalTo(Constants.ButtonSizes.Basics.height)
+            $0.width.greaterThanOrEqualTo(Constants.Button.basicSize.width)
```
Так задумка была чтобы вот так использовать можно было: $0.size.greaterThanOrEqualTo(Constants.Button.basicSize)

danildemchenko replied:
а если мне greaterThanOrEqualTo только для width нужен?

VAmpersand replied:
С этим ты прав, можно разделить, но у тебя разницы быть не должно. Ширина кнопки может зависеть от контента(длинный заголовка). А вот высота нет. Высота статична, по этому все должно быть нормально.

danildemchenko committed:
the additional task

1. implement general handler instead of three separate functions
2. fix TabBarController selectedIndex (previous homework, the changes disappeared somehow)

danildemchenko committed:
a little fix

size instead of width+height

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilCustomView.swift:
```diff
@@ -0,0 +1,78 @@
+
+final class DanilCustomView: UIView {
+    
+    enum ButtonName {
```
Тебе не нужен этот энум. Можешь просто название передать в (title: )

VAmpersand commented on Module_2 Groupe_1/Danil/TabController/DanilCustomView.swift:
```diff
@@ -0,0 +1,78 @@
+
+@objc extension DanilCustomView {
+    private func buttonHandler(_ sender: UIButton) {
+        switch sender.titleLabel?.text {
```
Тут сработает проверка по sender:

    private func buttonHandler(_ sender: UIButton) {
        switch sender {
        case callbackButton: changeBackgroundColorCallback(.red)
        case delegateButton: transferDataDelegate.changeBackgroundColor(to: .green)
        case notificationButton:  NotificationCenter.default.post(name: .changeColor, object: UIColor.blue)
        default: break
        }
    }

И кстати никогда не суй в 'default:' действия если оно действительно не нужно.

danildemchenko committed:
get rid of the redundant enum
